Robust_Lab.qmd

```diff
@@ -41,85 +47,295 @@ Use the `stars` data in `robustbase`. This data looks at the relationship betwee
 1.  
 
     ```{r}
+    #| output: false
     stars<-robustbase::starsCYG
     ```
 
     a\. Plot the data and describe the pattern seen. What is Pearson's *r*?
 
+    ```{r}
+    ggplot(stars, aes(x=log.Te,y=log.light)) +
+      geom_point() + geom_smooth()
+    cor(stars$log.Te, stars$log.light, method="pearson")
+    ```
+
+    There is a U-shaped curve whereby the extreme coldest and hottest stars have the greatest light intensity.
+
     b\. Re-run the correlation, but this time use the winsorized r (20%). Do this manually and then with the correlation::correlation function from `easystats`.
 
+    ```{r}
+    stars_win <- winsorize(stars, threshold=.2)
+    cor(stars_win$log.Te, stars_win$log.light)
+
+    kable(correlation::correlation(stars, winsorize=.2))
+
+    ```
+
     c\. Compare the correlations.
 
+    The winzorized correlation has a positive value (.34) compared to the negative correlation obtained without winsorizing (-.21), because the extreme values have been replaced with the nearest scores.
+
 ## Bootstrapping and Permutations
 
 2.  For the following data: \[8.453532, 10.025041, 11.495339, 9.367600, 8.333229, 9.788753, 10.883344, 10.543059, 9.869095, 10.799819\]
 
     a\. Bootstrap the mean (using the `boot` package) and plot the histogram with `ggplot2`
 
+    ```{r}
+    numbers <- c(8.453532, 10.025041, 11.495339, 9.367600, 8.333229, 9.788753, 10.883344, 10.543059, 9.869095, 10.799819)
+    set.seed(12)
+    boot_mean <- one.boot(numbers, mean, R=1000)
+    boot_mean_df <- as.data.frame(boot_mean$t)
+    ggplot() + geom_histogram(data=boot_mean_df,aes(V1)) +
+      labs(x='Means')
+
```

Comment:
one boot is fine to use but wanted you to get some practice with writing functions for mean and median here.